tools/graphiteindex/src/main/java/com/criteo/biggraphite/graphiteindex/LuceneIndexSearcher.java

```diff
@@ -0,0 +1,249 @@
+        throws IOException
+    {
+        this.baseCfs = baseCfs;
+        this.readCommand = readCommand;
+        this.column = column;
+        this.queryController = new QueryController(baseCfs,
+                (PartitionRangeReadCommand) readCommand,
+                DatabaseDescriptor.getRangeRpcTimeout());
+    }
+
+    /**
+     * TODO(p.boddu): We are searching and then creating the iterator. Incorporate searching in the iterator. This will
+     * save time from unneeded results (for e.g. limit 10)
+     *
+     * @param executionController
+     * @return
+     */
+    @Override
+    public UnfilteredPartitionIterator search(ReadExecutionController executionController) {
+        // TODO(p.boddu): Handle multiple expressions.
+        ByteBuffer indexValueBb = readCommand.rowFilter().getExpressions().get(0).getIndexValue();
+        String indexValue = UTF8Type.instance.compose(indexValueBb);
+        BooleanQuery query = patternToQuery(indexValue);
+        logger.debug("Searching Lucene index for column:{} with value:{} luceneQuery:{}",
+                column.name.toString(), indexValue, query);
+
+        Map<SSTableReader, List<Long>> searchResultsBySSTable =
+                baseCfs.getLiveSSTables().stream()
+                        .map(ssTable -> {
+                            String indexName = GraphiteSASI.makeIndexName(column, ssTable.descriptor.generation);
+                            Path indexPath = new File(ssTable.descriptor.directory, indexName).toPath();
+                            logger.debug("Searching index:{} with indexPath:{}", ssTable.getFilename(), indexPath);
+                            List<Long> offsets = searchOffsets(query, indexPath);
+                            return Pair.of(ssTable, offsets);})
+                        .collect(Collectors.toMap(t -> t.getLeft(), t -> t.getRight()));
+
+        // TODO(p.boddu): Remove this unnecessary tranformation.
+        List<Pair<SSTableReader, Long>> searchResults =
+                searchResultsBySSTable.entrySet()
+                        .stream()
+                        .flatMap(es -> es.getValue().stream()
+                                .map(offset -> Pair.of(es.getKey(), offset))
+                        ).collect(Collectors.toList());
+
+        return new LuceneResultsIterator(searchResults, readCommand, queryController, executionController);
+    }
+
+
+    private static class LuceneResultsIterator implements UnfilteredPartitionIterator {
+        private final ReadCommand readCommand;
+        private final QueryController queryController;
+        private final ReadExecutionController executionController;
+        private final List<Pair<SSTableReader, Long>> searchResults;
+        private final Iterator<Pair<SSTableReader, Long>> searchResultsIterator;
+
+        public LuceneResultsIterator(List<Pair<SSTableReader, Long>> searchResults,
```

Comment:
and you add some java doc for this constructor ?

Reply:
done